src/js/experiments/snippetFeedback.js

```diff
@@ -0,0 +1,261 @@
+      this.feedbackForm = container;
+      this.wasThisHelpfulContainer.appendChild(container);
+
+      // Here we setup the document event listeners that handle hiding the form
+      // when the user clicks outside of it.
+      //
+      // Why do we use a setTimeout with a timer of 0?
+      // -> Because there is a race condition with the adding of the click
+      // listeners to the document that immediately close the form when it opens
+      //
+      // Setting it to 0 causes the inner callback to be run asynchronously, adding it to the eventqueue
+      // so that it executes after the form being opened.
+      setTimeout(() => {
+        const outsideClickListener = (event) => {
+          if (!this.feedbackForm.contains(event.target)) {
+            this.wasThisHelpfulContainer.removeChild(this.feedbackForm);
+            this.feedbackForm = null;
+            removeClickListener();
+          }
+        };
+
+        const removeClickListener = () => {
+          document.removeEventListener('click', outsideClickListener);
+        };
+
+        document.addEventListener('click', outsideClickListener);
+      }, 0);
+    }
+  }
+
+  /**
+   * Render the char count and set it's result on the respective element.
+   * */
+  _renderCharCount() {
+    let charCount = this.currCharCount;
+    let charCountLimited =
+      charCount > SnippetFeedback.MAX_CHAR_COUNT
+        ? SnippetFeedback.MAX_CHAR_COUNT
+        : charCount;
+    const el = `
+        <div class="charCountRow">
+          <span>${charCountLimited} / ${SnippetFeedback.MAX_CHAR_COUNT}</span>
+        </div>`;
+    this.currCharCountElement.innerHTML = el;
+  }
+
+  // -- Trackers --
+  // These are responsible for sending data based on user input
+  // to amplitude.
+
+  /**
+   * _trackYesOrNoButton sends a trackAction to aplitude.
+   * actions are: "docs-snippet-helpful-no"  and "docs-snippet-helpful-yes"
+   * */
+  _trackYesOrNoButton(yesOrNoString) {
+    const date = new Date();
+    window.AnalyticsClient.trackAction(
+      `docs-snippet-helpful-${yesOrNoString}`,
+      {
+        originatingSnippet: this.snippetElement.textContent,
+        location: window.location.pathname,
+        timeOfButtonClick: date.toISOString(),
+      },
+    );
+  }
+
+  /**
+   * `_trackFormSubmission` invokes trackAction with a payload of the original
+   * snippet, the feedback of the user, and the timeOfFormSubmission.
+   * */
+  _trackFormSubmission(formContent) {
+    const date = new Date();
+    window.AnalyticsClient.trackAction(`docs-snippet-helpful-form-submission`, {
+      originatingSnippet: this.snippetElement.textContent,
+      feedback: formContent,
+      location: window.location.pathname,
+      wasThisHelpful: this.wasThisHelpful,
+      // for diffing this against the time that the _trackYesOrNoButton was clicked
+      // as well as possible the time the copy code button was clicked (TODO add that)
+      timeOfFormSubmission: date.toISOString(),
```

Comment:
Amplitude events already have a time associated with it so I don't think that is necessary:

![Screen Shot 2021-12-07 at 10 49 00 PM](https://user-images.githubusercontent.com/1449325/145161941-73e797fc-ab0a-4864-9966-f091d123b37f.png)